tests/test_geometry.py

```diff
@@ -1,190 +1,183 @@
+GEOMETRY1 = Geometry(shapely.geometry.Polygon([(0, 0), (0, 1), (1, 1), (1, 0), (0, 0)]), CRS(32633))
 GEOMETRY2 = Geometry(WKT_STRING, CRS.WGS84)
-BBOX = BBox(bbox=[14.00, 45.00, 14.03, 45.03], crs=CRS.WGS84)
+BBOX = BBox(bbox=(14.00, 45.00, 14.03, 45.03), crs=CRS.WGS84)
 
 GEOMETRY_LIST = [GEOMETRY1, GEOMETRY2, BBOX]
 
 
-def _round_point_coords(x: float, y: float, decimals: int = 1) -> Tuple[float, float]:
-    """Rounds coordinates of a point"""
-    return round(x, decimals), round(y, decimals)
-
-
-def test_bbox_no_crs() -> None:
-    with pytest.raises(TypeError):
-        BBox((46, 13, 47, 20))  # type: ignore[call-arg]
-
-
 @pytest.mark.parametrize(
-    "bbox_coords",
+    "coords, crs",
     [
-        [46.07, 13.23, 46.24, 13.57],
-        [46.24, 13.23, 46.07, 13.57],
-        [46.07, 13.57, 46.24, 13.23],
-        [46.24, 13.57, 46.07, 13.23],
+        ([[46.07, 13.23], [46.24, 13.57]], CRS.WGS84),
+        ((46.07, 13.23, 46.24, 13.57), CRS.POP_WEB),
+        (((46.07, 13.23), (46.24, 13.57)), CRS(8687)),
+        ([(46.07, 13.23), (46.24, 13.57)], CRS.WGS84),
+        ({"min_x": 46.07, "min_y": 13.23, "max_x": 46.24, "max_y": 13.57}, CRS.POP_WEB),
     ],
 )
-def test_bbox_from_flat_list(bbox_coords: List[float]) -> None:
-    bbox = BBox(bbox_coords, CRS.WGS84)
-    assert bbox.lower_left == (46.07, 13.23)
+def test_bbox_different_input_options(coords: Any, crs: CRS) -> None:
+    bbox = BBox(coords, crs)
     assert bbox.upper_right == (46.24, 13.57)
-    assert bbox.crs == CRS.WGS84
+    assert bbox.lower_left == (46.07, 13.23)
+    assert bbox.crs == crs
 
 
 @pytest.mark.parametrize(
-    "bbox_input",
+    "coords, crs",
     [
-        [[46.07, 13.23], [46.24, 13.57]],
-        (46.07, 13.23, 46.24, 13.57),
-        ((46.07, 13.23), (46.24, 13.57)),
-        [(46.07, 13.23), (46.24, 13.57)],
-        {"min_x": 46.07, "min_y": 13.23, "max_x": 46.24, "max_y": 13.57},
+        ({"x1": 46.07, "y1": 13.23, "x2": 46.24, "y2": 13.57}, CRS.WGS84),
+        ((46.07, 13.23, 46.24, 13.57), None),
+        ((46.07, 13.23, (46.24, 13.57)), CRS.WGS84),
     ],
 )
-def test_bbox_different_input(bbox_input: Any) -> None:
-    bbox = BBox(bbox_input, CRS.WGS84)
-    assert bbox.upper_right == (46.24, 13.57)
-    assert bbox.lower_left == (46.07, 13.23)
-    assert bbox.crs == CRS.WGS84
-
-
-def test_bbox_from_bad_dict() -> None:
-    bbox_dict = {"x1": 46.07, "y1": 13.23, "x2": 46.24, "y2": 13.57}
-    with pytest.raises(KeyError):
-        BBox(bbox_dict, CRS.WGS84)
+def test_bbox_bad_input_options(coords: Any, crs: CRS) -> None:
+    with pytest.raises((KeyError, ValueError)):
+        BBox(coords, crs)
 
 
 def test_bbox_to_str() -> None:
-    x1, y1, x2, y2 = 45.0, 12.0, 47.0, 14.0
-    crs = CRS.WGS84
-    expect_str = f"{x1},{y1},{x2},{y2}"
-    bbox = BBox(((x1, y1), (x2, y2)), crs)
-    assert str(bbox) == expect_str
+    bbox = BBox(((45.0, 12.0, 47.0, 14.0)), CRS.WGS84)
+    assert str(bbox) == "45.0,12.0,47.0,14.0"
```

Comment:
add test for `reverse` and not sorted input

Reply:
the `__str__` method is probably gonna get :fire: after this MR

not sure we should test unsorted input as it is kind of an 'implementation detail'. We never mention that we're doing it. I'm not sure it's even correct of us to sort input, perhaps a warning would be better.

Reply:
No need to test ash ;)